cpp/src/arrow/python/numpy-internal.h

```diff
@@ -76,16 +76,12 @@ static inline std::string GetNumPyTypeName(int npy_type) {
     TYPE_CASE(INT16, "int16")
     TYPE_CASE(INT32, "int32")
     TYPE_CASE(INT64, "int64")
-#if (NPY_INT64 != NPY_LONGLONG)
```

Comment:
For some reason (macro expansion?) these `#if`s wouldn't work correctly here, even though `NPY_INT64` is defined to `NPY_LONG`.

Reply:
Hmm, actually, that must be because `NPY_LONGLONG` is not a macro...